packages/astro/src/vite-plugin-build-css/index.ts

```diff
@@ -0,0 +1,151 @@
+
+import type { ResolveIdHook, LoadHook, RenderedChunk } from 'rollup';
+import type { Plugin as VitePlugin } from 'vite';
+
+import { STYLE_EXTENSIONS } from '../core/ssr/css.js';
+import { getViteResolve, getViteLoad } from './resolve.js';
+import { getViteTransform, TransformHook } from '../vite-plugin-astro/styles.js';
+import * as path from 'path';
+
+
+const PLUGIN_NAME = '@astro/rollup-plugin-build-css';
+
+// This is a virtual module that represents the .astro <style> usage on a page
+const ASTRO_STYLE_PREFIX = '@astro-inline-style';
+
+const ASTRO_PAGE_STYLE_PREFIX = '@astro-page-all-styles';
+
+const isCSSRequest = (request: string) => STYLE_EXTENSIONS.has(path.extname(request));
```

Comment:
Vite may sometimes append query params to URLs, e.g. `styles.css?direct`. Probably safest to make sure those are being handled as well.